src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -61,10 +61,10 @@ def get_access_info(self, module_name):
         user = None
         password = None
         cert_file = None
-        secure_monitoring_stack = bool(mgr.get_module_option_ex('cephadm',
-                                                                'secure_monitoring_stack',
-                                                                'false'))
-        if secure_monitoring_stack:
+        secure_monitoring_stack = mgr.get_module_option_ex('cephadm',
+                                                           'secure_monitoring_stack',
+                                                           'false')
+        if secure_monitoring_stack.lower() == 'true':
```

Comment:
I don't understand how this fixes the linked tracker. In the tracker it's complaining that `This Orchestrator does not support 'orch prometheus access info'`. This change seems to be around how we check if the secure monitoring stack is is turned on or not. I don't even see where it's trying to run the `access info` (it instead seems to be running a `get-credentials`) around here.

Reply:
@adk3798 I changed the code a little bit. Basically the BUG is that `bool('false')` was returning True so even when the `secure_monitoring_stack` was disabled the dashboard was calling the `orch prometheus access info`. This issue didn't show up on the cephadm testing because this method exists and in the worst case we will be calling it even when the `secure_monitoring_stack` is disabled. But when rook is in use the call fails as rook operator doesn't implement the `orch prometheus access info` call.

Reply:
okay, so the error message is just misleading but it's actually the `get-credentials` call failing because it's being called when the secure_monitoring_stack option isn't set? If so, this makes more sense to me.

Would we also want a general check the orchestrator backend is cephadm in this case? It seems like this whole section where it even checks the config option is only relevant in the case cephadm is the backend orchestrator.

Reply:
Yes, and that was happening because the rook module didn't have the call implemented at all. We didn't see that issue with cephadm because the api `orch cephadm get-credentials` exists. I'm not sure this only applies to `cephadm` .. in rook environments we could also have a prometheus which use authentication and in that case it would be useful to provide this info to the dashboard. 

@nizamial09 WDYT?

Reply:
The check is explicitly looking for the cephadm module's config option. Even if this got implemented in rook, we'd need to make a separate check for that. That's why I was thinking a check of what the backend orchestrator is makes sense.

Reply:
@adk3798 yeah, that's right. Even in that case this check is specific to cephadm.

Reply:
@rkachach is it possible to make this config option (`secure_monitoring_stack`) to deal with both rook and cephadm deployments? maybe something on the level of orchestrator? It doesn't makes sense now but when/if the rook deployment also has a secured monitoring stack, we could use that config to get the credentials, regardless of checking the backend orchestrator explicitly in the dashboard.

Reply:
@adk3798 @nizamial09 by now I'll just added a check for cephadm. In the near future we probably need to add the same support for rook as well.